salt/cloud/clouds/profitbricks.py

```diff
@@ -0,0 +1,836 @@
+# -*- coding: utf-8 -*-
+'''
+ProfitBricks Cloud Module
+=========================
+
+The ProfitBricks SaltStack cloud module allows a ProfitBricks server to
+be automatically deployed and bootstraped with Salt.
+
+:depends: profitbrick >= 2.3.0
+
+The module requires ProfitBricks credentials to be supplied along with
+an existing virtual datacenter UUID where the server resources will
+reside. The server should also be assigned a public LAN, a private LAN,
+or both along with SSH key pairs.
+...
+
+Set up the cloud configuration at ``/etc/salt/cloud.providers`` or
+``/etc/salt/cloud.providers.d/profitbricks.conf``:
+
+.. code-block:: yaml
+
+    my-profitbricks-config:
+      driver: profitbricks
+      # The ProfitBricks login username
+      username: user@example.com
+      # The ProfitBricks login password
+      password: secretpassword
+      # The ProfitBricks virtual datacenter UUID
+      datacenter_id: <UUID>
+      # SSH private key filename
+      ssh_private_key: /path/to/private.key
+      # SSH public key filename
+      ssh_public_key: /path/to/public.key
+      # Assign node to a specified public LAN.
+      public_lan: <ID>
+      # Assign node to a specified proviate LAN.
+      private_lan: <ID>
+
+To use a private IP for connecting and bootstrapping node:
+
+.. code-block:: yaml
+
+    my-profitbricks-config:
+      ssh_interface: private_lan
+
+'''
+
+# Import python libs
+from __future__ import absolute_import
+import copy
+import logging
+import pprint
+import time
+
+# Import salt libs
+import salt.utils
+import salt.config as config
+from salt.utils import namespaced_function
+from salt.exceptions import (
+    SaltCloudSystemExit,
+    SaltCloudExecutionFailure,
+    SaltCloudExecutionTimeout
+)
+
+# Import salt.cloud libs
+from salt.cloud.libcloudfuncs import *
+import salt.utils.cloud
+
+try:
+    from profitbricks.client import ProfitBricksService
+    from profitbricks.client import Server, NIC, Volume
+    HAS_PROFITBRICKS = True
+except ImportError:
+    HAS_PROFITBRICKS = False
+
+# Get logging started
+log = logging.getLogger(__name__)
+
+__virtualname__ = 'profitbricks'
+
+
+# Some of the libcloud functions need to be in the same namespace as the
```

Comment:
This driver doesn't depend on `libcloud`, so these namespace definitions shouldn't be needed.